thorlcr/master/thgraphmanager.cpp

```diff
@@ -60,10 +61,157 @@ class CJobManager : public CSimpleInterface, implements IJobManager, implements
     StringAttr          currentWuid;
     ILogMsgHandler *logHandler;
 
+    CJobMaster *getCurrentJob() { CriticalBlock b(jobCrit); return jobs.ordinality() ? &OLINK(jobs.item(0)) : NULL; }
     bool executeGraph(IConstWorkUnit &workunit, const char *graphName, const SocketEndpoint &agentEp);
     void addJob(CJobMaster &job) { CriticalBlock b(jobCrit); jobs.append(job); }
     void removeJob(CJobMaster &job) { CriticalBlock b(jobCrit); jobs.zap(job); }
 
+    class CThorDebugListener : public CSimpleInterfaceOf<IInterface>, implements IThreaded
+    {
+    protected:
+        CThreaded threaded;
+        unsigned port;
+        Owned<ISocket> sock;
+        CJobManager &mgr;
+    private:
+        volatile bool running;
+    public:
+        CThorDebugListener(CJobManager &_mgr) : threaded("CThorDebugListener", this), mgr(_mgr)
+        {
+            port = globals->getPropInt("DebugPort", 9877);
```

Comment:
minor: default should really be in portlist.h, called something like THOR_DEBUG_PORT